SC Evo Complete/SCEvo_Core.SC2Mod/Base.SC2Data/GameData/UpgradeData.xml

```diff
@@ -156,262 +156,188 @@
-        <EffectArrayTemplate Reference="Unit,^ParamId^,LifeArmor" Value="{DataCollection,^ParamId^,UpgradeInfoUnitLifeArmorPerLevel}"/>
         <EffectArrayTemplate Operation="Set" Reference="Actor,^ParamId^,LifeArmorIcon" Value="Assets\Textures\btn-upgrade-protoss-groundarmorlevel2.dds"/>
-        <EditorCategories value="Race:Human,UpgradeType:ArmorBonus"/>
-        <DataCollection value="TerranInfantryArmorsSCBW"/>
-        <BonusResourcePerLevel index="Minerals" value="75"/>
-        <BonusResourcePerLevel index="Vespene" value="75"/>
+        <ScoreAmount value="350"/>
+        <LeaderLevel value="2"/>
     </CUpgrade>
-    <CUpgrade id="ProtossGroundArmorsSCBW@Lvl3" parent="SCBW_Upgrade">
+    <CUpgrade id="ProtossGroundArmorsSCBW@Lvl3" parent="ProtossGroundArmorsSCBW">
+        <Name value="Upgrade/Name/ProtossGroundArmorsSCBW@Lvl3"/>
         <Icon value="Assets\Textures\btn-upgrade-protoss-groundarmorlevel3.dds"/>
-        <Race value="Terr"/>
-        <EffectArrayTemplate Reference="Unit,^ParamId^,LifeArmorLevel" Value="1"/>
-        <EffectArrayTemplate Reference="Unit,^ParamId^,LifeArmor" Value="{DataCollection,^ParamId^,UpgradeInfoUnitLifeArmorPerLevel}"/>
         <EffectArrayTemplate Operation="Set" Reference="Actor,^ParamId^,LifeArmorIcon" Value="Assets\Textures\btn-upgrade-protoss-groundarmorlevel3.dds"/>
-        <EditorCategories value="Race:Human,UpgradeType:ArmorBonus"/>
-        <DataCollection value="TerranInfantryArmorsSCBW"/>
-        <BonusResourcePerLevel index="Minerals" value="75"/>
-        <BonusResourcePerLevel index="Vespene" value="75"/>
+        <ScoreAmount value="500"/>
+        <LeaderLevel value="3"/>
     </CUpgrade>
     <CUpgrade id="ProtossGroundWeaponsSCBW" parent="SCBW_Upgrade">
-        <Flags index="IgnoreInvalidReferenceInTemplate" value="0"/>
-        <Icon value="Assets\Textures\btn-upgrade-protoss-groundweaponslevel1.dds"/>
-        <Race value="Terr"/>
-        <ScoreAmount value="200"/>
+        <Icon value="Assets\Textures\btn-upgrade-protoss-groundweaponslevel0.dds"/>
+        <Race value="BWPr"/>
         <EffectArrayTemplate Reference="Weapon,^ParamId^Weapon,Level" Value="1"/>
         <EffectArrayTemplate Reference="Effect,^ParamId^Weapon@Damage,Amount" Value="{DataCollection,^ParamId^,UpgradeInfoWeapon[0].DamagePerDice}"/>
-        <EditorCategories value="Race:Terran,UpgradeType:AttackBonus"/>
+        <EditorCategories value="Race:Protoss,UpgradeType:AttackBonus"/>
         <BonusResourcePerLevel index="Minerals" value="75"/>
         <BonusResourcePerLevel index="Vespene" value="75"/>
+        <BonusTimePerLevel value="30"/>
+        <EffectArray Reference="Effect,DragoonSCBWWeapon@Damage,AttributeBonus[Armored]" Value="0.5"/>
+        <EffectArray Reference="Effect,DragoonSCBWWeapon@Damage,ShieldBonus" Value="0.5"/>
+        <EffectArray Reference="Effect,DragoonSCBWWeapon@Damage,AttributeBonus[Light]" Value="-0.5"/>
```

Comment:
Medium damage gets +1.5 from data collection upgrade data. These adjustments are made because otherwise we would be over-compensating for the explosive/concussive upgrade problem. The goal is to round up all upgrades to +1 (since that is required to keep up with armor upgrades), so weapons with +2 or larger can behave like before. The same goes for upgrades below for factory units + wraith + devourer